pkg/cmd/codespace/create.go

```diff
@@ -32,7 +32,7 @@ type createOptions struct {
 	permissionsOptOut bool
 	devContainerPath  string
 	idleTimeout       time.Duration
-	retentionPeriod   time.Duration
```

Comment:
could you use a `*time.Duration` instead? using pointers is a common pattern for optional types

Reply:
I'm not sure if I can because the default argument of [`DurationVar` from the pflag library ](https://pkg.go.dev/github.com/spf13/pflag#DurationVar) is of type `time.Duration` so I can't pass in `nil`. I'm open to suggestions if you have an idea of how this could be refactored.

Reply:
This is true but the library provides what you need to define flag funcs for your own types: https://pkg.go.dev/github.com/spf13/pflag#Var

https://pkg.go.dev/github.com/spf13/pflag#Value

Reply:
Here's how we approach `*string` and `*bool` flag value receivers in the codebase already: https://github.com/cli/cli/blob/ec12d27db5925c1f99193518252a971372acb71f/pkg/cmdutil/flags.go#L12-L16

While a pointer receiver is a valid way of differentiating between explicit empty value vs. flag not being set by the user, @reybard makes a good point that a custom flag receiver can be defined. If dealing with pointers proves to be convoluted, do not hesitate to make a custom value that wraps an underlying `time.Duration` value in an object that also keeps track of whether the flag was explicitly set or not. 👍